### DISCUSSION TITLE
pyinstaller needs additional folders to work

### DISCUSSION THREAD

**Discussion by bigfooted** [Kitware/trame#320] [Q&A] [answered]
Hi, 
When trying pyinstaller (linux ubuntu) on a trame example I get an executable, but I always get these kind of error messages when I start them:
```
  File "aiohttp/web_urldispatcher.py", line 557, in __init__
    raise ValueError("Not a directory")
ValueError: Not a directory
```
and:
`ValueError: No directory exists at '/home/user/trame/examples/02_vuetify/dist/01_menu/trame_vuetify/module/serve'`

For instance when trying one of the examples in trame/examples/02_vuetify/
`pyinstaller --onedir 01_menu.py`

I can manually add the missing folders, in this case they are:
```
/home/user/trame/examples/02_vuetify/dist/01_menu/trame_vuetify/module/serve
/home/user/trame/examples/02_vuetify/dist/01_menu/trame_client/module/vue2-www
```
Sometimes more folders were necessary. The application seems to start correctly when I create the folders, but I am not sure if more is needed?

**Comment by jourdain** [comment, marked as answer]
When using PyInstaller you need to provide additional information so it can bundle the "non-python" part of each of the trame packages you are using. You can see an example [here](https://github.com/Kitware/trame-tauri/tree/master/examples/simple-cone) where you need to **`--collect-data`** for `trame_client`, `trame_vuetify` and `trame_vtk`.
